datarepo-clienttests/src/main/java/utils/KubernetesClientUtils.java

```diff
@@ -0,0 +1,264 @@
+    // the ones constructed programmatically above
+    kubeConfig = new KubeConfig(contextsList, clusters, usersList);
+    kubeConfig.setContext(server.clusterName);
+
+    // build the client object from the config
+    ApiClient client = ClientBuilder.kubeconfig(kubeConfig).build();
+
+    // set the global default client to the one created above because the CoreV1Api and AppsV1Api
+    // constructors get the client object from the global configuration
+    Configuration.setDefaultApiClient(client);
+
+    kubernetesClientCoreObject = new CoreV1Api();
+    kubernetesClientAppsObject = new AppsV1Api();
+  }
+
+  /**
+   * List all the pods in the given namespace, or in the whole cluster if the namespace is not
+   * specified (i.e. null or empty string).
+   *
+   * @param namespace to list the pods in
+   * @return list of Kubernetes pods
+   */
+  public static List<V1Pod> listPods(String namespace) throws ApiException {
+    V1PodList list;
+    if (namespace == null || namespace.isEmpty()) {
+      list =
+          kubernetesClientCoreObject.listPodForAllNamespaces(
+              null, null, null, null, null, null, null, null, null);
+    } else {
+      list =
+          kubernetesClientCoreObject.listNamespacedPod(
+              namespace, null, null, null, null, null, null, null, null, null);
+    }
+    return list.getItems();
+  }
+
+  /**
+   * List all the deployments in the given namespace, or in the whole cluster if the namespace is
+   * not specified (i.e. null or empty string).
+   *
+   * @param namespace to list the deployments in
+   * @return list of Kubernetes deployments
+   */
+  public static List<V1Deployment> listDeployments(String namespace) throws ApiException {
+    V1DeploymentList list;
+    if (namespace == null || namespace.isEmpty()) {
+      list =
+          kubernetesClientAppsObject.listDeploymentForAllNamespaces(
+              null, null, null, null, null, null, null, null, null);
+    } else {
+      list =
+          kubernetesClientAppsObject.listNamespacedDeployment(
+              namespace, null, null, null, null, null, null, null, null, null);
+    }
+    return list.getItems();
+  }
+
+  /**
+   * Get the API deployment in the given namespace, or in the whole cluster if the namespace is not
+   * specified (i.e. null or empty string). This method expects that there is a single API
+   * deployment in the namespace.
+   *
+   * @param namespace to get the API deployment from
+   * @return the API deployment, null if not found
+   */
+  public static V1Deployment getApiDeployment(String namespace) throws ApiException {
+    // loop through the deployments in the namespace
+    // find the one that matches the api component label
+    return listDeployments(namespace).stream()
+        .filter(
+            deployment ->
+                deployment.getMetadata().getLabels().get(componentLabel).equals(apiComponentLabel))
+        .findFirst()
+        .orElse(null);
```

Comment:
This is great -- much less brittle than trying to match the container name.

Reply:
Credit to @smark88 for explaining the labels piece at some point a while ago.